### PR TITLE
fix: `fully_qualified_strict_types` with `leading_backslash_in_global_namespace` enabled - handle reserved types in phpDoc

### PR DESCRIPTION
**Pull request by mvorisek**
missed in #5620, tests cherrypicked from #7623

It is super important to not prepend `\` to reserved phpdoc types like `int` otherwise `leading_backslash_in_global_namespace: true` is useless/broken.

### CODE REVIEW

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 1]
```diff
@@ -277,9 +277,9 @@ class SomeClass extends \Foo\Bar\A implements \Foo\Bar\Izumi, A, \A\B, \Foo\Bar\
         yield 'catch in multiple namespaces' => [
             '<?php
 namespace {
-    try{ foo(); } catch (Exception $z) {}
-    try{ foo(); } catch (A\X $z) {}
-    try{ foo(); } catch (B\Z $z) {}
+    try{ foo(); } catch (\Exception $z) {}
```
Comment (mvorisek) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:280]: this is an edge case, but I belive `leading_backslash_in_global_namespace: true` should prepend the (FQ)CN with `\`

File: tests/Fixtures/Integration/misc/fully_qualified_strict_types-leading_backslash_in_global_namespace.test [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+    "@PhpCsFixer": true,
+    "fully_qualified_strict_types": ["leading_backslash_in_global_namespace": true]
+}
+--EXPECT--
+<?php
+
+/**
+ * @param \DateTime $dateTime
+ * @param int $v
```
Comment (mvorisek) [inline tests/Fixtures/Integration/misc/fully_qualified_strict_types-leading_backslash_in_global_namespace.test:13]: It is super important to not prepend `\` otherwise `leading_backslash_in_global_namespace: true` is useless/broken.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php (hunk 1 above)
Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:280]: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.45.0/src/Fixer/Import/FullyQualifiedStrictTypesFixer.php#L432 should be `false ===` - I will commit fix shortly, introduced in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7459

Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:280]: addressed in #7654

**Comment by mvorisek** [comment]
@Wirone please reopen this PR, #7654 was meant to fix one issue/comment found in this PR, not the whole PR

**Comment by mvorisek** [comment]
Let's merge this small bugfix. I need it for another PR.

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @mvorisek 🍻!
